cdap-ui/app/features/adapters/controllers/detail-ctrl.js

```diff
@@ -15,12 +15,214 @@
+              namespace: $state.params.namespace,
+              appId: rAdapterDetail.name,
+              scheduleId: 'etlWorkflow',
+              scope: $scope
+            })
+              .$promise
+              .then(function (schedule) {
+                $scope.appStatus = schedule.status;
+              });
+          }
+        });
+
+    } else {
+
+      angular.forEach(rAdapterDetail.programs, function (program) {
+        if (program.type === 'Workers') {
+          params.workerId = program.id;
+        }
+      });
+
+      myWorkersApi.pollStatus(params)
+        .$promise
+        .then(function (res) {
+          $scope.appStatus = res.status;
+        });
+    }
+
+
+    $scope.do = function (action) {
+      var scheduleParams = {
+        namespace: $state.params.namespace,
+        appId: rAdapterDetail.name,
+        scheduleId: 'etlWorkflow',
+        scope: $scope
+      };
+
+      switch (action) {
+        case 'Start':
+          $scope.appStatus = 'Starting';
+          if (AdapterDetail.programType === 'WORKFLOWS') {
+            myWorkFlowApi.scheduleResume(scheduleParams, {})
+              .$promise
+              .then(function () {
+                $scope.appStatus = 'SCHEDULED';
+              });
+          } else {
+            myWorkersApi.doAction(angular.extend(params, { action: 'start' }, {}))
+              .$promise
+              .then(function () {
+                $scope.appStatus = 'RUNNING';
+              }, function () {
+                $scope.appStatus = 'FAILED';
+              });
+          }
+
+          break;
+
+        case 'Stop':
+          $scope.appStatus = 'Stopping';
+          if (AdapterDetail.programType === 'WORKFLOWS') {
+            myWorkFlowApi.scheduleSuspend(scheduleParams, {})
+              .$promise
+              .then(function () {
+                $scope.appStatus = 'SUSPENDED';
+              });
+          } else {
+            myWorkersApi.doAction(angular.extend(params, { action: 'stop' }, {}))
+              .$promise
+              .then(function () {
+                $scope.appStatus = 'STOPPED';
+              });
+          }
+
+          break;
+        case 'Run Once':
+          myWorkFlowApi.doAction(angular.extend(params, { action: 'start' }), {});
+
+          break;
+        case 'Delete':
+          var deleteParams = {
+            namespace: $state.params.namespace,
+            appId: rAdapterDetail.name,
+            scope: $scope
+          };
+
+          myAppsApi.delete(deleteParams)
+            .$promise
+            .then(function () {
+              console.log('Successfully Deleted Hydrator App');
+              $state.go('adapters.list');
+            });
+
+          break;
+      }
+    };
+
+
+    $scope.nodes = [];
+
+    function initializeDAG() {
+      try{
+        rAdapterDetail.config = JSON.parse(rAdapterDetail.configuration);
+      } catch(e) {
+        console.log('ERROR in configuration from backend: ', e);
+      }
```

Comment:
Should we have a return statement here? Beyond this catch block nothing is valid right?